packages/salesforcedx-utils-vscode/src/telemetry/telemetry.ts

```diff
@@ -79,12 +79,12 @@ export class TelemetryService {
   public async initializeService(
     extensionContext: ExtensionContext,
     extensionName: string,
-    aiKey: string,
-    version: string
+    version: string,
```

Comment:
Howdy @floralan 
What you think about making the following update? 

```suggestion
    extensionContext: ExtensionContext
  ): Promise<void> {
    const { name, version, aiKey } = extensionContext.extension.packageJSON;
    this.extensionContext = extensionContext;
    this.extensionName = name;
```
This would allow us to avoid syncing the change across extensions (the extra parameters would just be ignored), but would require some more touches in this repo? I took a look and it appears most places with couple of exceptions where name is hard coded are doing the pull from the packageJSON property on the context anyway so this make this more consistent across extensions. 

It would probably be worthwhile to also add some defensive coding to verify that the expected JSON properties are found and throw a clear error if that's not that case.

Reply:
note this would remove all parameters but the context.